src/sage/interfaces/sagespawn.pyx

```diff
@@ -199,9 +199,12 @@ class SagePtyProcess(PtyProcess):
                     # but the default (UTF-8) will do now, since I don't
                     # think we have any non-ASCII quit_strings anyways.
                     self.write(str_to_bytes(self.quit_string))
-                except (OSError, IOError):
+                except (OSError, IOError, PtyProcessError):
                     pass
-            self.fileobj.close()
+            try:
+                self.fileobj.close()
+            except (OSError, IOError, PtyProcessError):
+                pass
```

Comment:
What is this for?

Reply:
here is what one gets without this change on macOS. On Linux these two commands complete just fine.
```
sage: gap.eval('quit;')
''
sage: gap.quit()
---------------------------------------------------------------------------
OSError                                   Traceback (most recent call last)
Cell In [2], line 1
----> 1 gap.quit()

File ~/software/sage/src/sage/interfaces/expect.py:640, in Expect.quit(self, verbose)
    638         else:
    639             print("Exiting %r" % (self._expect,))
--> 640     self._close()
    641 self._reset_expect()

File ~/software/sage/src/sage/interfaces/expect.py:564, in Expect._close(self, force)
    562 try:
    563     if self._expect is not None:
--> 564         self._expect.close(force=force)
    565 except (PtyProcessError, ExceptionPexpect):
    566     self._expect.ptyproc.fd = -1

File /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/pexpect/pty_spawn.py:328, in spawn.close(self, force)
    324 self.flush()
    325 with _wrap_ptyprocess_err():
    326     # PtyProcessError may be raised if it is not possible to terminate
    327     # the child.
--> 328     self.ptyproc.close(force=force)
    329 self.isalive()  # Update exit status from ptyproc
    330 self.child_fd = -1

File ~/software/sage/src/sage/interfaces/sagespawn.pyx:204, in sage.interfaces.sagespawn.SagePtyProcess.close()
    202     except (OSError, IOError):
    203         pass
--> 204 self.fileobj.close()
    205 self.fd = -1
    206 self.closed = 1

OSError: [Errno 5] Input/output error
sage: 
```